File: 240307/마라톤 중간에 택시타기 2/taking-a-taxi-in-the-middle-of-the-marathon-2.py
import sys

INT_MAX = sys.maxsize

n = int(input())
arr = [list(map(int, input().split())) for _ in range(n)]

ans = INT_MAX
for i in range(1, n - 1):
	dist = 0
	prev_idx = 0
	for j in range(1, n):
		if j == i:
			continue
		dist += abs(arr[prev_idx][0] - arr[j][0]) + abs(arr[prev_idx][1] - arr[j][1])
		prev_idx = j
	
	ans = min(ans, dist)


print(ans)

File: 240307/마라톤 중간에 택시타기 2/taking-a-taxi-in-the-middle-of-the-marathon-2.js
const fs = require('fs')
const [n, ...arr] = fs.readFileSync(0).toString().trim().split('\n')

const N = +n
const points = arr.map(row => row.trim().split(' ').map(Number))
let minDis = Number.MAX_SAFE_INTEGER

for (let i=1;i<N-1;i++) {
    let prev = 0
    let sum = 0
    for (let j=1;j<N;j++) {
        if (i === j) continue

        sum += Math.abs(points[j][0] - points[prev][0]) + Math.abs(points[j][1] - points[prev][1])
        prev = j
    }

    minDis = Math.min(minDis, sum)
}

console.log(minDis)